File: tool/conformance.dart
import 'dart:convert';

import 'package:wallet_hd_derivation_kit/wallet_hd_derivation_kit.dart';

void main() {
  const words =
      'abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about';
  final result = <String, String>{};
  for (final chain in supportedChains()) {
    result[chain.id] = deriveAddress(
      source: const {'mnemonic': words},
      chain: chain.id,
    )['address']! as String;
  }
  print('WALLETHD_CONFORMANCE=${jsonEncode(result)}');
}

File: rust/src/bin/conformance.rs
use std::collections::BTreeMap;
use wallet_hd_derivation_kit::{derive_address, DeriveOptions, Source, CHAINS};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let words = "abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about";
    let source = Source::mnemonic(words, "");
    let values = CHAINS
        .iter()
        .map(|chain| {
            let value = derive_address(
                &source,
                DeriveOptions {
                    chain: chain.id,
                    ..Default::default()
                },
            )?;
            Ok((chain.id, value.address))
        })
        .collect::<Result<BTreeMap<_, _>, wallet_hd_derivation_kit::HdError>>()?;
    println!("WALLETHD_CONFORMANCE={}", serde_json::to_string(&values)?);
    Ok(())
}
